trl/trainer/iterative_sft_config.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass, field
+from typing import Optional
+
+from ..core import flatten_dict
+
+
+@dataclass
+class IterativeSFTConfig(object):
```

Comment:
note that we ported the CLI args to `tyro` since you opened the PR. not a big change, you can for example look at https://github.com/huggingface/trl/blob/main/trl/trainer/ppo_config.py

Reply:
I have removed the `IterativeSFTConfig` as it was not useful anymore as we inherit from `Trainer`